atomix/cluster/src/test/java/io/atomix/cluster/messaging/impl/NettyMessagingServiceTest.java

```diff
@@ -729,4 +712,48 @@ public void testNoRemoteHandlerExceptionEmptyStringValue() {
         .isInstanceOf(MessagingException.NoRemoteHandler.class)
         .withMessage(expectedException.getMessage());
   }
+
+  @EnabledOnOs(OS.LINUX)
+  @RegressionTest("https://github.com/camunda/zeebe/issues/14837")
+  void shouldNotLeakUdpSockets() throws IOException {
+    // given
+    // the configured amount of threads for Netty's epoll transport
+    final var maxConnections = Runtime.getRuntime().availableProcessors() * 2;
+    final var subject = nextSubject();
+    netty2.registerHandler(
+        subject, (address, bytes) -> new byte[0], MoreExecutors.directExecutor());
+
+    // when
+    for (int i = 0; i < maxConnections * 4; i++) {
+      netty1.sendAndReceive(netty2.address(), subject, "hello world".getBytes(), false).join();
+    }
+
+    // then - there seems to be a slight amount more than maxConnections normally, but without the
+    // fix this was way, way, way more, so it should be fine to allow a little bit more than the
+    // expected max number of connections
+    assertThat(udpSocketCount()).isLessThanOrEqualTo(maxConnections * 2L);
```

Comment:
I'm not happy that we have to account for more, but if you remove the patch, you'll see that the number of open sockets is way, way more than the max connections, so it's still correct. But I wish we could have an accurate number =/